src/plugins/data/public/query/filter_manager/lib/compare_filters.ts

```diff
@@ -44,9 +44,9 @@ const mapFilter = (
 ) => {
   const cleaned: FilterMeta = omit(filter, excludedAttributes);
 
-  if (comparators.negate) cleaned.negate = filter.meta && Boolean(filter.meta.negate);
-  if (comparators.disabled) cleaned.disabled = filter.meta && Boolean(filter.meta.disabled);
-  if (comparators.disabled) cleaned.alias = filter.meta?.alias;
+  if (comparators.negate) cleaned.negate = filter?.meta && Boolean(filter?.meta.negate);
```

Comment:
Since according to TypeScript, `filter` can't be null, there's no need to add the optional chaining operator here.
It's also quite an implicit way to check for this.

Instead, please add another case to `compareFilters` where if one of the passed filters is null, it returns false.

Reply:
Ah, that makes sense. I just pushed a commit that removed the optional chaining and added a null check for first and second filters at the top of `compareFilters`. 

The line I wrote (shown at the bottom of comment) explicitly checks for a `null` value instead of writing `if(!first || !second)` though because I wasn't sure if other falsy values like an empty string or `0` should return false as well. If you'd prefer `if(!first || !second)` though I'd be happy to update 👍 

`if (first === null || second === null) return false;`

Reply:
I think `if(!first || !second)` would be more correct.
Otherwise looks great! Thank you!

Reply:
Sounds good! Just pushed a commit to use `if(!first || !second)` instead.